File: dragonsquad/craneam/com/UI.java
package dragon.craneam.com;

import java.util.ArrayList;
import java.util.List;
import android.util.Log;

public class UI
  {
    private static final String TAG = "Dragon Squad";  // Game tag in logs
    private static final boolean DEBUG = true;    // Debug flag
    private Buffers graphicsBuffer;
    public List<Text> textFields = new ArrayList<Text>();

    public UI()
      {
        graphicsBuffer = new Buffers();
        try
          {
            textFields.clear();
          }
        catch (UnsupportedOperationException e)
          {
            if (DEBUG)
              {
                Log.e(TAG, "UI: Error clearing text fields");
              }
          }
      }

    public void AddGraphicElement(UIElement element)
      {
        element.AddIndices(graphicsBuffer);
        element.AddVertexes(graphicsBuffer);
        element.AddTextureCoordinates(graphicsBuffer);
      }

    public void AddTextElement(Text element)
      {
        try
          {
            textFields.add(element);
            if (DEBUG)
              {
                Log.d(TAG, "UI: Added text field succesfully.");
              }
          }
        catch (UnsupportedOperationException e)
          {
            if (DEBUG)
              {
                Log.e(TAG, "UI: Error adding text element: " + e);
              }
          }
        catch (ClassCastException e)
          {
            if (DEBUG)
              {
                Log.e(TAG, "UI: Error adding text element: " + e);
              }
          }
        catch (IllegalArgumentException e)
          {
            if (DEBUG)
              {
                Log.e(TAG, "UI: Error adding text element: " + e);
              }
          }
      }

    public Buffers GetGraphicsBuffer()
      {
        return graphicsBuffer;
      }
  }

File: app/src/main/java/com/koodipuukko/dragonsquad/UI.kt
package com.koodipuukko.dragonsquad

import java.util.ArrayList
import android.util.Log

class UI {
	private val graphicsBuffer: Buffers = Buffers()
	var textFields: MutableList<Text> = ArrayList<Text>()

	init {
		try {
			textFields.clear()
		} catch(e: UnsupportedOperationException) {
			if(DEBUG) {
				Log.e(TAG, "UI: Error clearing text fields")
			}
		}
	}

	fun addGraphicElement(element: UIElement) {
		element.addIndices(graphicsBuffer)
		element.addVertexes(graphicsBuffer)
		element.addTextureCoordinates(graphicsBuffer)
	}

	fun addTextElement(element: Text?) {
		try {
			if(element != null) {
				textFields.add(element)
				if(DEBUG) {
					Log.d(TAG, "UI: Added text field succesfully.")
				}
			}
		} catch(e: UnsupportedOperationException) {
			if(DEBUG) {
				Log.e(TAG, "UI: Error adding text element: $e")
			}
		} catch(e: ClassCastException) {
			if(DEBUG) {
				Log.e(TAG, "UI: Error adding text element: $e")
			}
		} catch(e: IllegalArgumentException) {
			if(DEBUG) {
				Log.e(TAG, "UI: Error adding text element: $e")
			}
		}
	}

	fun getGraphicsBuffer(): Buffers {
		return graphicsBuffer
	}

	companion object {
		private const val TAG = "Dragon Squad" // Game tag in logs
		private const val DEBUG = true // Debug flag
	}
}
